test/test.js

```diff
@@ -10,8 +13,31 @@ jasmine.loadConfig({
     ]
 });
 
-var runner =  new JsReporters.JasmineAdapter(jasmine);
-var testReporter = new JsReporters.TestReporter(runner, referenceData.Jasmine);
+var jasmineRunner = new JsReporters.JasmineAdapter(jasmine.env);
+jasmine.addReporter({}); // Suppress the default reporter
+var jasmineTestReporter = new JsReporters.TestReporter(jasmineRunner, referenceData.Jasmine);
 
 jasmine.execute();
-process.exit(testReporter.ok ? 0 : 1);
+if (!jasmineTestReporter.ok) {
+    throw new Error("Tests for JasmineAdapter failed!");
+}
+
+//TODO: Use a proper test framework?
+var done = new Promise(function (resolve) {
+    QUnit.done(function () {
+        resolve();
+    });
+});
+
+var qunitRunner = new JsReporters.QUnitAdapter();
+var qunitTestReporter = new JsReporters.TestReporter(qunitRunner, referenceData.QUnit);
+
+QUnit.config.autorun = false;
+
+require("./qunit/tests.js");
+
+QUnit.load();
+
+done.then(function () {
+    process.exit(qunitTestReporter.ok ? 0 : 1);
```

Comment:
An unhandled exception doesn't cause a non zero exit code? That's surprising, sounds like a bug in nodejs.